src/openrct2/management/NewsItem.cpp

```diff
@@ -207,73 +207,72 @@ static int32_t news_item_get_new_history_slot()
  *
  *  rct2: 0x0066BA74
  */
-void news_item_get_subject_location(int32_t type, int32_t subject, int32_t* x, int32_t* y, int32_t* z)
+CoordsXYZ news_item_get_subject_location(int32_t type, int32_t subject)
 {
-    Ride* ride;
-    Peep* peep;
-    Vehicle* vehicle;
+    CoordsXYZ subjectLoc;
 
     switch (type)
     {
         case NEWS_ITEM_RIDE:
-            ride = get_ride(subject);
+        {
+            Ride* ride = get_ride(subject);
             if (ride == nullptr || ride->overall_view.isNull())
             {
-                *x = LOCATION_NULL;
+                subjectLoc.setNull();
                 break;
             }
-            *x = ride->overall_view.x + 16;
-            *y = ride->overall_view.y + 16;
-            *z = tile_element_height({ *x, *y });
+            auto rideViewCentre = ride->overall_view.ToTileCentre();
+            subjectLoc = CoordsXYZ{ rideViewCentre, tile_element_height(rideViewCentre) };
             break;
+        }
         case NEWS_ITEM_PEEP_ON_RIDE:
-            peep = GET_PEEP(subject);
-            *x = peep->x;
-            *y = peep->y;
-            *z = peep->z;
-            if (*x != LOCATION_NULL)
+        {
+            Peep* peep = GET_PEEP(subject);
+            subjectLoc = CoordsXYZ{ peep->x, peep->y, peep->z };
+            if (!subjectLoc.isNull())
                 break;
 
             if (peep->state != 3 && peep->state != 7)
             {
-                *x = LOCATION_NULL;
+                subjectLoc.setNull();
                 break;
             }
 
             // Find which ride peep is on
-            ride = get_ride(peep->current_ride);
+            Ride* ride = get_ride(peep->current_ride);
             if (ride == nullptr || !(ride->lifecycle_flags & RIDE_LIFECYCLE_ON_TRACK))
             {
-                *x = LOCATION_NULL;
+                subjectLoc.setNull();
                 break;
             }
 
             // Find the first car of the train peep is on
-            vehicle = &(get_sprite(ride->vehicles[peep->current_train])->vehicle);
+            Vehicle* vehicle = &(get_sprite(ride->vehicles[peep->current_train])->vehicle);
```

Comment:
can you change the  `&(get_sprite(ride->vehicles[peep->current_train])->vehicle)` to `GET_VEHCILE(ride->vehicles[peep->current_train])`. I'm going to change it anyway in a latter refactor but this would look neater.